ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -516,8 +516,13 @@ public ReplicaId next() {
    * @return {@code true} if not found count in originating datacenter exceeds threshold, {@code false} otherwise.
    */
   private boolean hasFailedOnOriginatingDcNotFound() {
-    return originatingDcNotFoundFailureThreshold > 0
-        && originatingDcNotFoundCount >= originatingDcNotFoundFailureThreshold;
+    // Check that there are at least routerConfig.routerPutSuccessTarget number of replicas in leader or standby state
+    // in originating data center before failing the request. Also, in case originating data center no longer exists,
+    // avoid checking for this condition.
+    return originatingDcNotFoundFailureThreshold > 0 && (originatingDcNotFoundCount
+        >= originatingDcNotFoundFailureThreshold) && (this.originatingDcName != null
+        && getEligibleReplicas(this.originatingDcName, EnumSet.of(ReplicaState.STANDBY, ReplicaState.LEADER)).size()
+        >= routerConfig.routerPutSuccessTarget);
```

Comment:
why it's put target? this is used mostly in get? we can make this change in the SimpleOperationTracker's constructor instead of in this method. So in the constructor we query for the number of eligible replicas in the originating dc, and if it's less than the target, we change the originatingDcNotFoundFailureThreshold to 0, so we disable this feature.

Reply:
Using the `getEligibleReplicas(originatingDcName, EnumSet.of(ReplicaState.STANDBY, ReplicaState.LEADER)).size()
 >= routerConfig.routerPutSuccessTarget)` to make sure that there is at least one or more replicas from the original PUT quorum in leader or standby state. I was wondering that when we receive not-found response from all replicas, this might help us to be sure that blob is missing. 

Sure. Moved the logic to constructor as suggested.